File: 2015/src/test/java/nl/sbmf21/aoc15/days/Day11Test.java
package nl.sbmf21.aoc15.days;

import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.HashMap;

import static nl.sbmf21.aoc.testing.UtilKt.buildWithInput;
import static nl.sbmf21.aoc.testing.UtilKt.testDay;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class Day11Test {

    @Test
    public void testInput() {
        testDay(Day11.class, "cqjxxyzz", "cqkaabcc", false, null);
    }

    @Test
    public void testExample() {
        Day11 day = buildWithInput(Day11.class, new ArrayList<>());

        new HashMap<String, String>() {{
            put("abcdefgh", "abcdffaa");
            put("ghijklmn", "ghjaabcc");
        }}.forEach((input, expected) -> assertEquals(expected, day.nextValid(input)));
    }
}

File: 2021/kotlin/src/test/kotlin/nl/sbmf21/aoc21/days/Day11Test.kt
package nl.sbmf21.aoc21.days

import nl.sbmf21.aoc.testing.testDay
import kotlin.test.Test

class Day11Test {

    @Test
    fun testInput() = testDay(Day11::class.java, 1721, 298)

    @Test
    fun testExample() = testDay(Day11::class.java, 1656, 195, true)
}